.github/workflows/ci-cd-workflow.yml

```diff
@@ -196,11 +194,12 @@ jobs:
           wait-for-service-stability: true
 
       - name: Push stable tag to ECR
-        env:
-          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
-          ECR_REPOSITORY: ${{ github.event.repository.name }}
-        run: |
-          docker push $ECR_REGISTRY/$ECR_REPOSITORY:stable
+        uses: tinact/docker.image-retag@master
+        with:
+          image_name: ${{ steps.login-ecr.outputs.registry }}/${{ github.event.repository.name }}:${{ github.sha }}
+          image_old_tag: latest
```

Comment:
Latest won't always be the tag we want to target (concurrent builds), we'd need to use the commit sha here.

Does this action remove the old tag?

Have you checked in to how its done to make sure its not just doing `docker tag` like we were or doing a `docker pull` which would be easier and clearer for us to do ourselves.

Reply:
It will Retags a given Docker Image and pushs the new tag to the given registry.

Reply:
Considering it just does a pull, tag and push we can do that ourselves to keep the workflow file clear. There is another approach on the original comment that updates the remote image without pulling.

Reply:
Have pushed again and added `docker pull` before tagging.